csharp/test/Apache.Arrow.Tests/Extensions/Net472Extensions.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Collections.Generic;
+
+namespace Apache.Arrow.Tests
+{
+    internal static class Net472Extensions
+    {
+        public static IEnumerable<(TFirst First, TSecond Second)> Zip<TFirst, TSecond>(this IEnumerable<TFirst> first, IEnumerable<TSecond> second)
+        {
+            using (var enumerator1 = first.GetEnumerator())
+            using (var enumerator2 = second.GetEnumerator())
+            {
+                while (enumerator1.MoveNext() && enumerator2.MoveNext())
```

Comment:
Should we detect and throw an exception if the iterators are uneven (e.g. `enumerator1.MoveNext()` returns `false` but `enumerator2.MoveNext()` returns `true`?

Reply:
From what I can tell, this is the behavior of the Zip extension method in the newer BCL. (I can leave an explicit comment to that effect, but thought the name of the class was a sufficient hint :). )

Reply:
No need for a comment.  I recognized it was meant to be a polyfill I just didn't realize what the old behavior was.  However, I agree with you that this matches the behavior described in https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable.zip?view=net-7.0 so we can resolve this.